python/pyiceberg/expressions/base.py

```diff
@@ -867,3 +875,196 @@ def visit_unbound_predicate(self, predicate) -> BooleanExpression:
 
     def visit_bound_predicate(self, predicate) -> BooleanExpression:
         return predicate
+
+
+ROWS_MIGHT_MATCH = True
+ROWS_CANNOT_MATCH = False
+IN_PREDICATE_LIMIT = 200
+
+
+def _from_byte_buffer(field_type: IcebergType, val: bytes):
+    if not isinstance(field_type, PrimitiveType):
+        raise ValueError(f"Expected a PrimitiveType, got: {type(field_type)}")
+    return from_bytes(field_type, val)
+
+
+class ManifestEvaluator(BoundBooleanExpressionVisitor[bool]):
```

Comment:
It's okay to combine these, but it does seem a little strange to expose so much in the public `ManifestEvaluator`. I somewhat prefer using a separate `_ManifestEvalVisitor` class.

Reply:
Yes, I went a bit back and forth on this. I don't like having a setter that sets something very internal. I now created a function that returns a Callable, that hides the visitor:
```python
def manifest_evaluator(
    schema: Schema, partition_filter: UnboundPredicate, case_sensitive: bool = True
) -> Callable[[ManifestFile], bool]:
    evaluator = _ManifestEvalVisitor(schema, partition_filter, case_sensitive)
    return evaluator.eval
```